Dockerfile

```diff
@@ -1,11 +1,37 @@
-FROM golang:1.18 AS builder
+FROM --platform=${BUILDPLATFORM} golang:1.21 AS builder
 
-WORKDIR /go/src/github.com/apecloud/open-local
-COPY . .
-RUN /bin/sh -c 'pwd && make build && chmod +x bin/open-local'
+## docker buildx build injected build-args:
+#BUILDPLATFORM — matches the current machine. (e.g. linux/amd64)
+#BUILDOS — os component of BUILDPLATFORM, e.g. linux
+#BUILDARCH — e.g. amd64, arm64, riscv64
+#BUILDVARIANT — used to set ARM variant, e.g. v7
+#TARGETPLATFORM — The value set with --platform flag on build
+#TARGETOS - OS component from --platform, e.g. linux
+#TARGETARCH - Architecture from --platform, e.g. arm64
+#TARGETVARIANT
 
-FROM sting2me/open-local-base:latest
+ARG TARGETOS
+ARG TARGETARCH
+
+ARG GOPROXY=https://proxy.golang.org,direct
+
+WORKDIR /src
+
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
+RUN --mount=type=bind,target=. \
+    --mount=type=cache,target=/go/pkg/mod \
+    go mod download -x
+
+RUN --mount=type=bind,target=. \
+    --mount=type=cache,target=/root/.cache/go-build \
+    --mount=type=cache,target=/go/pkg/mod \
+    go env && \
+    GOOS=${TARGETOS} GOARCH=${TARGETARCH} OUTPUT_DIR=/out make build
+
+FROM alpine:3.18
```

Comment:
use this image is not recommended?

Reply:
in sfs csi, I used registry.access.redhat.com/ubi8/ubi:8.8

Reply:
Let's do this in other PR. ref #10